### PR TITLE
feat: update user and group pages design

### PR DESCRIPTION
**Pull request by andre-code**
Introduce a refreshed design for the user and group pages.

/deploy

### CODE REVIEW

**Review by Panaetius — APPROVED** [review]

File: CHANGELOG.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ User-Facing Changes
 
 - **UI**: Simplify the creation of PolyBox and SwitchDrive data connectors (`#3396 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3396>`__).
 - **UI**: Simplify the project and group creation interactions in Renku 2.0 to a simple modal (`#3399 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3399>`__).
-- **UI**: Introduce a refreshed design for the dashboard in Renku 2.0 (`#3407 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3407>`__).
+- **UI**: Introduce a refreshed design for the dashboard, user, and group pages in Renku 2.0 (`#3407 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3407>`__).
```
Comment (leafty) [inline CHANGELOG.rst:37]: Maybe add the PR here?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -61,11 +61,12 @@ Individual Components
 - `renku-data-services 0.28.0 <https://github.com/SwissDataScienceCenter/renku-data-services/releases/tag/v0.28.0>`_
 - `renku-search 0.7.0 <https://github.com/SwissDataScienceCenter/renku-search/releases/tag/v0.7.0>`_
 - `renku-ui 3.43.0 <https://github.com/SwissDataScienceCenter/renku-ui/releases/tag/3.43.0>`_
+- `renku-ui 3.44.0 <https://github.com/SwissDataScienceCenter/renku-ui/releases/tag/3.43.0>`_
```
Comment (leafty) [inline CHANGELOG.rst:64]: wrong link
```suggestion
- `renku-ui 3.44.0 <https://github.com/SwissDataScienceCenter/renku-ui/releases/tag/3.44.0>`_
```

**Comment by leafty** [comment]
Note: can we also bump to 3.44.1 here? -> https://github.com/SwissDataScienceCenter/renku-ui/releases/tag/3.44.1

File: CHANGELOG.rst (hunk 1 above)
Comment (leafty) [reply CHANGELOG.rst:37]:
```suggestion
- **UI**: Introduce a refreshed design for the dashboard, user, and group pages in Renku 2.0 (`#3407 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3407>`__, `#3428 <https://github.com/SwissDataScienceCenter/renku-ui/pull/3428>`__).
```

**Review by Panaetius — APPROVED** [review]

**Pull request merged**